src/main/java/characterpicker/infrastructure/CharacterParser.java

```diff
@@ -3,26 +3,55 @@
 import characterpicker.domain.Character;
 import org.apache.commons.lang.StringUtils;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
 public class CharacterParser {
-    private CharacterParser(){
+    private CharacterParser() {
         throw new IllegalStateException("Utility class");
     }
 
     static Character parseCharacter(String unparsedCharacter) {
 
-        String[] characterProperties = unparsedCharacter.split(",");
+        if (unparsedCharacter.endsWith(";")){
+            throw new IllegalArgumentException("Completion required. If you don't have one type 'none'");
+        }
+
+        String[] unparsedProperties = unparsedCharacter.split(";");
```

Comment:
Trochę mi nie pasuje nazwa `unparsedProperties` bo nie informuje czytelnika kodu, że tutaj chodzi o tego charactera i completion marki.

Ale nie mam na razie pomysłu na inną nazwę. I może nie ma co bo zakładam, że zostanie to zastąpione podawniem konfiguracji completion marków w pliku i może problem sam się rozwiąże.